iphone/Classes/PlatformModule.m

```diff
@@ -297,17 +297,17 @@ - (NSNumber *)totalMemory
 }
 GETTER_IMPL(NSNumber *, availableMemory, AvailableMemory);
 
-- (BOOL)openURL:(NSString *)url withOptions:(id)options andCallback:(JSValue *)callback
+- (BOOL)openURL:(NSString *)url withOptions:(JSValue *)options andCallback:(JSValue *)callback
 {
   NSURL *newUrl = [TiUtils toURL:url proxy:self];
   BOOL result = NO;
 
-  // iOS 10+
   NSMutableDictionary *optionsDict = [NSMutableDictionary dictionary];
-  if ([options isKindOfClass:[NSDictionary class]]) {
-    optionsDict = (NSMutableDictionary *)options;
-  } else if ([options isKindOfClass:[JSValue class]]) {
-    callback = (JSValue *)options;
+  NSDictionary *dictionary = [options toDictionary];
+  if (dictionary.allKeys.count > 0) {
```

Comment:
Although highly unlikely, but something like the following would break this check:

```js
const callback = () => {};
callback.foo = 'bar';
Ti.Platform.openURL('https://www.google.com', {}, callback);
```

Unfortunately, JSValue has no built-in functionality to test whether a value is a function or not. However, the C-API has [JSObjectIsFunction](https://developer.apple.com/documentation/javascriptcore/1451769-jsobjectisfunction?language=objc), which we could use for an extension:

```objc
@interface JSValue (TitaniumAdditions)

- (BOOL)isFunction;

@end

@implementation JSValue (TitaniumAdditions)

- (BOOL)isFunction {
  if (!self.isObject) {
    return false;
  }
  
  JSContextRef context = self.context.JSGlobalContextRef;
  JSObjectRef object = JSValueToObject(context, self.JSValueRef, NULL);
  if (object == NULL) {
    return NO;
  }
  return JSObjectIsFunction(context, object);
}

@end
```

Reply:
Good idea. Updated PR according to suggestion.